cli/js/tests/headers_test.ts

```diff
@@ -276,9 +281,9 @@ unitTest(function headerParamsArgumentsCheck(): void {
       }
     }
     assertEquals(hasThrown, 2);
-    assertEquals(
+    assertStrContains(
       errMsg,
-      `Headers.${method} requires at least 1 argument, but only 0 present`
+      `${method} requires at least 1 argument, but only 0 present`
```

Comment:
@kitsonk ... with this change `Headers.name == HeadersImpl`.

I'm sure we can workaround this somehow - but it's not immediately obvious to me how.

Reply:
I think in the case of HeadersImpl, it could be done in dom_iterable.ts
https://github.com/denoland/deno/blob/8397cd52a5ec915a0b5e85d337cfcc38f4aaa63f/cli/js/web/dom_iterable.ts#L82-L86
however there are other classes, like URLSearchParams which would also have to be fixed.

Reply:
Were it is meaningful, we just need to override it.  I wouldn't put it in `dom_iterable` because that overloads what it does it... just something like the below and throw it in util somewhere:

```ts
export function setName(fn: Function, value: string): void {
  Object.defineProperty(fn, "name", { value, configurable: true };
}
```

Reply:
Yeah that would work... it is more overhead than just a normal class definition (not much, admittedly, and maybe none after snapshotting). Ultimately we're just working around our own TypeScript (dis)organization - ideally the JS that gets executed would just be 
```
class Header {
  // ...
}
```
not 
```
class HeaderImpl {
  // ..
}
Object.defineProperty(HeaderImpl, "name", { "Header", configurable: true };
```

I don't have a good solution on how to get our TS to produce the optimal JS :/